Modules/_testclinic.c

```diff
@@ -0,0 +1,959 @@
+#ifndef Py_BUILD_CORE_BUILTIN
+#  define Py_BUILD_CORE_MODULE 1
+#endif
+
+/* Always enable assertions */
+#undef NDEBUG
+
+#define PY_SSIZE_T_CLEAN
+
+#include "Python.h"
+
+#include "clinic/_testclinic.c.h"
+
+
+/* Pack arguments to a tuple, implicitly increase all the arguments' refcount.
+ * NULL arguments will be replaced to Py_None. */
+static PyObject *
+pack_arguments_newref(int argc, ...)
+{
+    assert(!PyErr_Occurred());
+    PyObject *tuple = PyTuple_New(argc);
+    if (!tuple) {
+        return NULL;
+    }
+
+    va_list vargs;
+    va_start(vargs, argc);
+    for (int i = 0; i < argc; i++) {
+        PyObject *arg = va_arg(vargs, PyObject *);
+        if (arg) {
+            if (_PyObject_IsFreed(arg)) {
+                PyErr_Format(PyExc_AssertionError,
+                             "argument %d at %p is freed or corrupted!",
+                             i, arg);
+                va_end(vargs);
+                Py_DECREF(tuple);
+                return NULL;
+            }
+        }
+        else {
+            arg = Py_None;
+        }
+        PyTuple_SET_ITEM(tuple, i, Py_NewRef(arg));
+    }
+    va_end(vargs);
+    return tuple;
+}
+
+/* Pack arguments to a tuple, all the arguments' refcounts should be increased before passing in.
+ * Do not accept NULL arguments unless error occurs. */
+#define PACK_ARGUMENTS(argc, ...) do { \
```

Comment:
The naming should tell that this function performs an implicit `return` (see similar named macros `Py_RETURN_NONE`, `Py_RETURN_TRUE` and `Py_RETURN_FALSE`).

```suggestion
#define RETURN_PACKED_ARGS(argc, ...) do { \
```